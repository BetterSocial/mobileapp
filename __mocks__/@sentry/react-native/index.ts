export const addBreadcrumb = jest.fn();
