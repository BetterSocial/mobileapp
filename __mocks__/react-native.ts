import * as RN from 'react-native';

RN.Animated.timing = () => ({
    start: () => jest.fn()
})

module.exports = RN