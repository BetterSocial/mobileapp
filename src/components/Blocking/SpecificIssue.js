import * as React from 'react';
import IconFA5 from 'react-native-vector-icons/FontAwesome5';
import RBSheet from 'react-native-raw-bottom-sheet';
import {ActivityIndicator, StyleSheet, Text, TextInput, TouchableOpacity, View} from 'react-native';

import {Button} from '../Button';
import {fonts} from '../../utils/fonts';
import {COLORS} from '../../utils/theme';

const SpecificIssue = ({refSpecificIssue, onPress, onSkip, loading}) => {
  const [message, setMessage] = React.useState('');
  return (
    <RBSheet
      ref={refSpecificIssue}
      closeOnDragDown={true}
      closeOnPressMask={true}
      customStyles={{
        container: styles.container,
        draggableIcon: styles.draggableIcon
      }}>
      <View>
        <Text style={styles.title}>Please specify the issue</Text>
        <TextInput
          style={styles.input}
          value={message}
          onChangeText={(v) => setMessage(v)}
          textAlignVertical="top"
          multiline
          placeholder={'Please provide more details to inform our\n team (min. 50 characters)'}
          placeholderTextColor={COLORS.gray400}
        />
        <TouchableOpacity testID="button-skip-test" style={styles.btnSkip} onPress={() => onSkip()}>
          <Text style={styles.btnSkipText}>Skip & just block this account</Text>
          <IconFA5 name="chevron-right" size={17} color={COLORS.black} />
        </TouchableOpacity>
        <View style={styles.containerBtn}>
          <Button testID="button-report-test" onPress={() => onPress(message)}>
            {loading ? (
              <ActivityIndicator testID="loading-indicator-test" />
            ) : (
              <Text style={styles.containerBtnText}>File Report</Text>
            )}
          </Button>
        </View>
      </View>
    </RBSheet>
  );
};

export default SpecificIssue;

const styles = StyleSheet.create({
  title: {
    fontFamily: fonts.inter[700],
    fontSize: 18,
    color: COLORS.black,
    marginLeft: 21,
    marginTop: 18
  },
  containerBtn: {
    marginRight: 22,
    marginLeft: 18,
    marginBottom: 19
  },
  containerBtnText: {
    marginRight: 22,
    marginLeft: 18,
    marginBottom: 19,
    fontFamily: fonts.inter[500],
    fontSize: 16,
    color: COLORS.white
  },
  btn: {
    paddingLeft: 18,
    paddingRight: 22,
    paddingTop: 8
  },
  btnSkip: {
    backgroundColor: COLORS.gray100,
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingHorizontal: 17,
    paddingVertical: 18,
    marginVertical: 22
  },
  btnSkipText: {
    fontFamily: fonts.inter[700],
    fontSize: 14,
    color: COLORS.black
  },
  input: {
    backgroundColor: COLORS.gray100,
    marginLeft: 17,
    marginRight: 23,
    borderRadius: 8,
    height: 233,
    marginTop: 14,
    paddingVertical: 19,
    paddingRight: 13,
    paddingLeft: 19,
    fontFamily: fonts.inter[400],
    color: COLORS.white,
    borderWidth: 1,
    borderColor: COLORS.gray200
  },
  container: {
    height: 'auto',
    borderTopRightRadius: 20,
    borderTopLeftRadius: 20,
    backgroundColor: COLORS.almostBlack
  },
  draggableIcon: {
    backgroundColor: COLORS.gray100,
    width: 60
  }
});
