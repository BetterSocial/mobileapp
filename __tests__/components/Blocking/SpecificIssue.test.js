import * as React from 'react';
import renderer from 'react-test-renderer';
import {cleanup, fireEvent, render} from '@testing-library/react-native';

import SpecificIssue from '../../../src/components/Blocking/SpecificIssue';

jest.useFakeTimers();

describe('SpecificIssue Block Component test should pass', () => {
  afterAll(cleanup);
  it('should render', () => {
    const rbSheetOpenRefMock = jest.fn();
    const onPressMock = jest.fn();
    const onSkipMock = jest.fn();
    const loading = false;

    const ref = {
      current: {
        open: rbSheetOpenRefMock
      }
    };

    const tree = renderer
      .create(
        <SpecificIssue
          onPress={onPressMock}
          onSkip={onSkipMock}
          refSpecificIssue={ref}
          loading={loading}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render with loading', () => {
    const rbSheetOpenRefMock = jest.fn();
    const onPressMock = jest.fn();
    const onSkipMock = jest.fn();
    const loading = true;

    const ref = {
      current: {
        open: rbSheetOpenRefMock
      }
    };

    const {findByTestId} = render(
      <SpecificIssue
        onPress={onPressMock}
        onSkip={onSkipMock}
        refSpecificIssue={ref}
        loading={loading}
      />
    );
    expect(findByTestId('loading-indicator-test')).toBeTruthy();
  });

  it('should render with loading', async () => {
    const rbSheetOpenRefMock = jest.fn();
    const onPressMock = jest.fn();
    const onSkipMock = jest.fn();
    const loading = true;

    const ref = {
      current: {
        open: rbSheetOpenRefMock
      }
    };
    const realUseState = React.useState;
    const stubInitialState = [''];
    const mockState = jest
      .spyOn(React, 'useState')
      .mockImplementationOnce(() => realUseState(stubInitialState));

    const {getByTestId} = render(
      <SpecificIssue
        onPress={onPressMock}
        onSkip={onSkipMock}
        refSpecificIssue={ref}
        loading={loading}
      />
    );
    await fireEvent.changeText(getByTestId('changeText'));
    await fireEvent.press(getByTestId('button-skip-test'));
    await fireEvent.press(getByTestId('button-report-test'));
    expect(onSkipMock).toHaveBeenCalled();
    expect(onPressMock).toHaveBeenCalled();
  });
});
