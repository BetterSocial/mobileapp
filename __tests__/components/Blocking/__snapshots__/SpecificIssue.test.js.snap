// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpecificIssue Block Component test should pass should render 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  supportedOrientations={
    Array [
      "portrait",
      "portrait-upside-down",
      "landscape",
      "landscape-left",
      "landscape-right",
    ]
  }
  transparent={true}
  visible={false}
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Array [
          Object {
            "backgroundColor": "#00000077",
            "flex": 1,
          },
          undefined,
        ],
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
          "opacity": 1,
        }
      }
    />
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "backgroundColor": "#fff",
          "borderTopLeftRadius": 20,
          "borderTopRightRadius": 20,
          "height": "auto",
          "overflow": "hidden",
          "transform": Array [
            Object {
              "translateX": 0,
            },
            Object {
              "translateY": 0,
            },
          ],
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#ccc",
                "borderRadius": 5,
                "height": 5,
                "margin": 10,
                "width": 35,
              },
              Object {
                "backgroundColor": "#F5F6F7",
                "width": 60,
              },
            ]
          }
        />
      </View>
      <View>
        <Text
          style={
            Object {
              "color": "#1E1F20",
              "fontFamily": "Inter-Bold",
              "fontSize": 18,
              "marginLeft": 21,
              "marginTop": 18,
            }
          }
        >
          Please specify the issue
        </Text>
        <TextInput
          allowFontScaling={true}
          multiline={true}
          onChangeText={[Function]}
          placeholder="Please provide more details to inform our
 team (min. 50 characters)"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "#F5F6F7",
              "borderRadius": 8,
              "color": "#828282",
              "fontFamily": "Inter-Regular",
              "height": 233,
              "marginLeft": 17,
              "marginRight": 23,
              "marginTop": 14,
              "paddingLeft": 19,
              "paddingRight": 13,
              "paddingVertical": 19,
            }
          }
          textAlignVertical="top"
          underlineColorAndroid="transparent"
          value=""
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#F5F6F7",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginVertical": 22,
              "opacity": 1,
              "paddingHorizontal": 17,
              "paddingVertical": 18,
            }
          }
          testID="button-skip-test"
        >
          <Text
            style={
              Object {
                "color": "#1E1F20",
                "fontFamily": "Inter-Bold",
                "fontSize": 14,
              }
            }
          >
            Skip & just block this account
          </Text>
          <Icon
            color="#1E1F20"
            name="chevron-right"
            size={17}
          />
        </View>
        <View
          style={
            Object {
              "marginBottom": 19,
              "marginLeft": 18,
              "marginRight": 22,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#0391FB",
                "borderRadius": 6.4,
                "flexDirection": "row",
                "height": 40,
                "justifyContent": "center",
                "opacity": 1,
                "paddingHorizontal": 20,
              }
            }
            testID="button-report-test"
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "Inter-SemiBold",
                  "fontSize": 18,
                }
              }
            >
              <Text>
                File Report
              </Text>
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
